plugins/carla/CMakeLists.txt

```diff
@@ -0,0 +1,111 @@
+cmake_minimum_required(VERSION 3.16...3.25)
+
+legacy_check()
+
+option(ENABLE_CARLA "Enable building OBS with carla plugin host" ON)
+if(NOT ENABLE_CARLA)
+  target_disable(carla)
+  return()
+endif()
+
+# Find Qt
+find_qt(COMPONENTS Core Widgets)
+
+# Use pkg-config to find optional deps
+find_package(PkgConfig)
+
+# Find pthread via cmake
+if(OS_WINDOWS)
+  set(carla_pthread_libs OBS::w32-pthreads)
+else()
+  find_package(Threads REQUIRED)
+  set(carla_pthread_libs ${CMAKE_THREAD_LIBS_INIT})
+endif()
+
+# Optional: transient X11 window flags
+if(PKGCONFIG_FOUND AND NOT (OS_MACOS OR OS_WINDOWS))
+  pkg_check_modules(X11 "x11")
+else()
+  set(X11_FOUND FALSE)
+endif()
+
+# Use *.mm files under macOS, regular *.cpp everywhere else
+if(OS_MACOS)
+  set(CARLA_OBJCPP_EXT "mm")
+else()
+  set(CARLA_OBJCPP_EXT "cpp")
+endif()
+
+# Import extra carla libs
+include(cmake/jackbridge.cmake)
+add_library(carla::jackbridge ALIAS carla-jackbridge)
```

Comment:
The `add_library` call should happen inside the module itself, not outside of it. The included cmake list should generate it.

Also probably better to have these modules (if they're necessary in the first place) be included as proper subdirectories with their own `CMakeLists.txt` (if they're actually independent pieces), or alternatively create an `INTERFACE` type library if just the source code is needed and it's easier to bundle their usage requirements within a target.

Reply:
no longer relevant after carla side doing the main tool building, marking as resolved